contracts/xGNANABribes.sol

```diff
@@ -0,0 +1,155 @@
+ * Twitter:         https://twitter.com/ape_swap
+ * Telegram:        https://t.me/ape_swap
+ * Announcements:   https://t.me/ape_swap_news
+ * Reddit:          https://reddit.com/r/ApeSwap
+ * Instagram:       https://instagram.com/ApeSwap.finance
+ * GitHub:          https://github.com/ApeSwapFinance
+ */
+
+import "./xGNANA.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+/**
+    // TODO: Possible features
+    - [x] AccessControl to be able to whitelist IERC20VotesDelegatee from factory contracts
+    - [ ] Factory contract which can deploy bribe contracts with the callback functions below 
+    - [ ] Relevant events
+    - [ ] NatSpec comments
+    - [ ] Some ability to accumulate non GNANA revenue
+          - I would say that my argument still applies that we should perform buy backs at strategic points when there is high value
+             Is there a way we could mix in like stable revenue and with GNANA for xGNANA holders? 
+
+    - [x] Provide an admin role to be able to add new white listers
+    - [x] Be able to remove delegatee contracts
+ */
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+interface IERC20VotesDelegatee is IERC165 {
+    function syncDelegatePower(address delegator) external;
+}
+
+/**
+ * The intent here is that a factory contract could be used to deploy bridge contracts
+ * where a protocol can fill up with tokens and distribute tokens to xGNANA holders who 
+ * delegate their voting power.
+ */
+interface IxTokenWithBribes {
+    event CallbackDelegateRegistered(IERC20VotesDelegatee indexed delegatee);
+    event CallbackDelegateRemoved(IERC20VotesDelegatee indexed delegatee);
+    function syncDelegateVotingPower(address delegator, address delegatee, bool withRevert) external;
+    function registerCallbackDelegatee(IERC20VotesDelegatee delegateeContract) external;
+    function removeCallbackDelegatee(IERC20VotesDelegatee delegateeContract) external;
+    function getVotesDelegatedTo(address delegator, address delegatee) external returns (uint256);
+}
+
+contract xGNANABribes is IxTokenWithBribes, xGNANA, AccessControlEnumerable {
+    // NOTE: example https://github.com/ApeSwapFinance/apeswap-pool-factory/blob/f24fb874a9f70cedcfadeecd4a6ddacb45ee9752/contracts/PoolManager.sol
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    EnumerableSet.AddressSet private _registeredCallbackDelegatees;
+
+    bytes32 public constant DELEGATEE_MANAGER_ROLE = keccak256("DELEGATEE_MANAGER_ROLE");
+    
+    constructor(ERC20 gnana_, address admin_) xGNANA(gnana_) {
+        _grantRole(DEFAULT_ADMIN_ROLE, admin_);
+        _setRoleAdmin(DELEGATEE_MANAGER_ROLE, DEFAULT_ADMIN_ROLE);
+    }
+
+    function syncDelegateVotingPower(address delegator, address delegatee, bool withRevert) public {
+        bool synced = _syncDelegateVotingPower(delegator, delegatee);
+        if(withRevert && !synced) {
```

Comment:
Should probably turn this `internal` and provide a `public` function which automatically reverts if the `delegatee` is not registered.